src/calliope/backend/pyomo_backend_model.py

```diff
@@ -201,6 +203,16 @@ def _objective_setter(
 
         self._add_component(name, objective_dict, _objective_setter, "objectives")
 
+    def set_objective(self, name: str) -> None:  # noqa: D102, override
+        for obj_name, obj in self.objectives.items():
+            if obj.item().active and obj_name != name:
+                self.log("objectives", obj_name, "Objective deactivated.", level="info")
+                obj.item().deactivate()
+            if obj_name == name:
+                obj.item().activate()
+                self.log("objectives", obj_name, "Objective activated.", level="info")
+        self.objective = name
+
```

Comment:
I'd consider directly querying the requested name and using sets to deactivate the rest to reduce condition checks and to 'fail fast'.

Less checks, but same strength.

```suggestion
    def set_objective(self, name: str) -> None:  # noqa: D102, override
        self.objectives[name].activate()
        self.objective = name
        self.log("objectives", name, "Objective activated.", level="info")
        for key in self.objectives.keys() - {name}:
            if self.objectives[key].active:
                self.objectives[key].deactivate()
                self.log("objectives", key, "Objective deactivated.", level="info")
```